app/src/main/java/com/tamollyking/virtualsequins/MainActivity.kt

```diff
@@ -31,6 +31,7 @@ class MainActivity : AppCompatActivity() {
                 .build()
         val layoutMod = moshi.adapter<LayoutModel>(LayoutModel::class.java).fromJson(layoutJson)
         var sequinMap : Map<String, SequinModel> = HashMap<String, SequinModel>()
+        assertNotNull(layoutMod)
```

Comment:
You need to just do this in a way that's thread-safe.

Reply:
With Collections.synchronizedList we would have to manually synchronize on the list object before we iterate over it, whereas with CopyOnWriteArrayList creates a snapshot of the list at the time you iterate through it, so interference is impossible and the iterator is guaranteed not to throw ConcurrentModificationException

Reply:
Ok great. Thanks for the info!